From: Jozsef Kadlecsik <kadlec@blackhole.kfki.hu>
Date: Wed, 11 Feb 2009 13:00:35 +0000 (+0100)
Subject: Use the new hash really.
X-Git-Tag: v2.4.9~1
X-Git-Url: http://git.netfilter.org/cgi-bin/gitweb.cgi?p=ipset.git;a=commitdiff_plain;h=31c93a74639077ee39b448ff47f446b4864cfaea

Use the new hash really.

References to the old include file replaced with new one in order to
really use the new Jenkins' hash function.
---

diff --git a/kernel/ChangeLog b/kernel/ChangeLog
index 1ed0fbb..1c6e650 100644
--- a/kernel/ChangeLog
+++ b/kernel/ChangeLog
@@ -1,3 +1,7 @@
+2.4.9
+  - References to the old include file replaced with new one in order to
+    really use the new Jenkins' hash function.
+
 2.4.8
   - The Jenkins' hash lookup2() replaced with Jenkins' faster/better lookup3()
     hash function.
diff --git a/kernel/ip_set.c b/kernel/ip_set.c
index f60a63e..f52647f 100644
--- a/kernel/ip_set.c
+++ b/kernel/ip_set.c
@@ -19,7 +19,7 @@
 #include <linux/ip.h>
 #include <linux/skbuff.h>
 #include <linux/random.h>
-#include <linux/jhash.h>
+#include <linux/netfilter_ipv4/ip_set_jhash.h>
 #include <linux/errno.h>
 #include <linux/capability.h>
 #include <asm/uaccess.h>
diff --git a/kernel/ip_set_iphash.c b/kernel/ip_set_iphash.c
index b1bccc1..aac3eec 100644
--- a/kernel/ip_set_iphash.c
+++ b/kernel/ip_set_iphash.c
@@ -11,7 +11,7 @@
 #include <linux/moduleparam.h>
 #include <linux/ip.h>
 #include <linux/skbuff.h>
-#include <linux/jhash.h>
+#include <linux/netfilter_ipv4/ip_set_jhash.h>
 #include <linux/errno.h>
 #include <asm/uaccess.h>
 #include <asm/bitops.h>
diff --git a/kernel/ip_set_ipporthash.c b/kernel/ip_set_ipporthash.c
index 2352858..c41c0a8 100644
--- a/kernel/ip_set_ipporthash.c
+++ b/kernel/ip_set_ipporthash.c
@@ -13,7 +13,7 @@
 #include <linux/tcp.h>
 #include <linux/udp.h>
 #include <linux/skbuff.h>
-#include <linux/jhash.h>
+#include <linux/netfilter_ipv4/ip_set_jhash.h>
 #include <linux/errno.h>
 #include <asm/uaccess.h>
 #include <asm/bitops.h>
diff --git a/kernel/ip_set_ipportiphash.c b/kernel/ip_set_ipportiphash.c
index e0725d9..166bec4 100644
--- a/kernel/ip_set_ipportiphash.c
+++ b/kernel/ip_set_ipportiphash.c
@@ -13,7 +13,7 @@
 #include <linux/tcp.h>
 #include <linux/udp.h>
 #include <linux/skbuff.h>
-#include <linux/jhash.h>
+#include <linux/netfilter_ipv4/ip_set_jhash.h>
 #include <linux/errno.h>
 #include <asm/uaccess.h>
 #include <asm/bitops.h>
diff --git a/kernel/ip_set_ipportnethash.c b/kernel/ip_set_ipportnethash.c
index 267370a..fe36c58 100644
--- a/kernel/ip_set_ipportnethash.c
+++ b/kernel/ip_set_ipportnethash.c
@@ -13,7 +13,7 @@
 #include <linux/tcp.h>
 #include <linux/udp.h>
 #include <linux/skbuff.h>
-#include <linux/jhash.h>
+#include <linux/netfilter_ipv4/ip_set_jhash.h>
 #include <linux/errno.h>
 #include <asm/uaccess.h>
 #include <asm/bitops.h>
diff --git a/kernel/ip_set_nethash.c b/kernel/ip_set_nethash.c
index 268fe6a..27267d9 100644
--- a/kernel/ip_set_nethash.c
+++ b/kernel/ip_set_nethash.c
@@ -11,7 +11,7 @@
 #include <linux/moduleparam.h>
 #include <linux/ip.h>
 #include <linux/skbuff.h>
-#include <linux/jhash.h>
+#include <linux/netfilter_ipv4/ip_set_jhash.h>
 #include <linux/errno.h>
 #include <asm/uaccess.h>
 #include <asm/bitops.h>
